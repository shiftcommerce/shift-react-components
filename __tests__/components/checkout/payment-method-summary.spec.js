// Libraries
import React from 'react'

// Components
import PaymentMethodSummary from '../../../src/components/checkout/payment-method-summary'
import PaymentMethodHeader from '../../../src/components/checkout/payment-method-header'

// Objects
import Image from '../../../src/objects/image'

test("renders the shipping methods summary", () => {
  // Arrange
  const paymentMethod = 'credit-card'

  // Act
  const wrapper = shallow(<PaymentMethodSummary paymentMethod={paymentMethod} headerTitle={'test'} showEditButton={true}/>)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('div').first()).toMatchSelector('.c-payment-method__summary')
  expect(wrapper.find('div').last()).toMatchSelector('.c-payment-method__summary-information')
  expect(wrapper.find(PaymentMethodHeader).length).toEqual(1)
  expect(wrapper).toIncludeText(paymentMethod)
})


test('renders the PayPal image and does not render edit button', () => {
  // Arrange
  const paymentMethod = 'PayPal'

  // Act
  const wrapper = shallow(<PaymentMethodSummary paymentMethod={paymentMethod} headerTitle={'test'} showEditButton={false}/>)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('div').first()).toMatchSelector('.c-payment-method__summary')
  expect(wrapper.find('div').last()).toMatchSelector('.c-payment-method__summary-information')
  expect(wrapper.find(PaymentMethodHeader).length).toEqual(1)
  expect(wrapper.find(Image).length).toEqual(1)
})
