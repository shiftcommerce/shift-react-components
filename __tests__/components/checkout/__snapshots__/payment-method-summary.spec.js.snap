// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PayPal image and does not render edit button 1`] = `
<div
  className="c-payment-method__summary"
>
  <PaymentMethodHeader
    collapsed={true}
    showEditButton={false}
    title="test"
  />
  <div
    className="c-payment-method__summary-information"
  >
    <Image
      className="c-payment-method__summary-information-card-image"
      src="/static/payments/pay-pal.svg"
    />
  </div>
</div>
`;

exports[`renders the shipping methods summary 1`] = `
<div
  className="c-payment-method__summary"
>
  <PaymentMethodHeader
    collapsed={true}
    showEditButton={true}
    title="test"
  />
  <div
    className="c-payment-method__summary-information"
  >
    <p>
      credit-card
    </p>
  </div>
</div>
`;
